### ISSUE TITLE
Stabilise `nix store ping`

### ISSUE THREAD

**Issue by fricklerhandwerk** [NixOS/nix#8914]
This implements https://github.com/NixOS/rfcs/pull/136. This issue is agreed-upon by the https://github.com/orgs/NixOS/teams/nix-team

Required changes:

- rename to `nix store info`
- return bool or null for the `trusted` field instead of currently int
  - if the underlying `enum` is converted to an `enum struct`, the potential to make this mistake will dissapear 
- ensure that `--json` output fits the [JSON guidelines](https://nixos.org/manual/nix/unstable/contributing/cli-guideline#returning-future-proof-json)

**Comment by vicky1999** [comment]
Hello, I would like to work on this task.  Can I start working on this?

**Comment by Ericson2314** [comment]
Yes you can!

**Comment by vicky1999** [comment]
@fricklerhandwerk 
I have a doubt in point 2:
Based on the comment, I have made the changes to display the true/false value for trusted value in store info and info json.  Kindly let me know if this is the expectation. 

![image](https://github.com/NixOS/nix/assets/24916780/33a5cece-27c3-40aa-9d63-a95af407ff33)

**Comment by vicky1999** [comment]
@fricklerhandwerk The json payload, I saw that the json argument is not formatted.   This will be fine.  But, I think it will look more nicer if the json is formatted.  So, I have made some changes to display the formatted json instead of single line json. 

![image](https://github.com/NixOS/nix/assets/24916780/0f424428-5d64-4474-9152-addaf5bb763e)

Kindly let me know if I can push this change to the current open PR (#8975 ).
